library/src/main/java/com/google/android/exoplayer/util/BitArray.java

```diff
@@ -151,7 +151,7 @@ public int readUnsignedByte() {
     byte b;
```

Comment:
Oh yeah, you're right. It should be:

```
public int readUnsignedByte() {
    int value;
    if (bitOffset != 0) {
      value = ((data[byteOffset] & 0xFF) << bitOffset)
          | ((data[byteOffset + 1] & 0xFF) >>> (8 - bitOffset));
    } else {
      value = data[byteOffset];
    }
    byteOffset++;
    return value & 0xFF;
  }
```

By the way - We'll be pushing some tests to Github soon. So when that happens, we should make sure this class is unit tested.